library/systemd/src/lib/yast2/system_service.rb

```diff
@@ -0,0 +1,490 @@
+          :on_demand
+        else
+          :manual
+        end
+    end
+
+    # Whether the service is currently active in the system
+    #
+    # @return [Boolean]
+    def current_active?
+      service.active? || socket_active?
+    end
+
+    # Returns the list of supported start modes
+    #
+    # * :on_boot:   The service will be started when the system boots.
+    # * :on_demand: The service will be started on demand.
+    # * :manual:    The service is disabled and it will be started manually.
+    #
+    # @return [Array<Symbol>] List of supported modes.
+    def start_modes
+      return @start_modes if @start_modes
+      @start_modes = [:on_boot, :manual]
+      @start_modes << :on_demand if socket
+      @start_modes
+    end
+
+    # Returns the start mode
+    #
+    # See {#start_modes} to find out the supported modes for a given service (usually :on_boot,
+    # :manual and, in some cases, :on_demand).
+    #
+    # @note This is a temporary value (not saved yet). Use {#current_start_mode} to get the actual
+    #   start mode of the service in the system.
+    #
+    # @return [Symbol] :on_boot, :on_demand, :manual
+    def start_mode
+      new_value_for(:start_mode) || current_start_mode
+    end
+
+    # Sets the service start mode
+    #
+    # See {#start_modes} to find out the supported modes for a given service (usually :on_boot,
+    # :manual and, in some cases, :on_demand). The given value will be applied after calling {#save}.
+    #
+    # @see #start_modes
+    #
+    # @raise ArgumentError when mode is not valid
+    def start_mode=(mode)
+      if !start_modes.include?(mode)
+        raise ArgumentError, "Invalid start mode: '#{mode}' for service '#{service.name}'"
+      end
+
+      if mode == current_start_mode
+        unregister_change(:start_mode)
+      else
+        register_change(:start_mode, mode)
+      end
+    end
+
+    # Whether the service supports :on_demand start mode
+    #
+    # @return [Boolean]
+    def support_start_on_demand?
+      start_modes.include?(:on_demand)
+    end
+
+    # Whether the service will be active after calling {#save}
+    #
+    # @note This is a temporary value (not saved yet). Use {#current_active?} to get the actual
+    #   active value of the service in the system.
+    #
+    # @return [Boolean] true if the service must be active; false otherwise
+    def active?
+      return new_value_for(:active) if changed?(:active)
+      current_active?
+    end
+
+    # Keywords to search for this service
+    #
+    # In case the service has an associated socket, the socket name
+    # is included as keyword.
+    #
+    # @return [Array<String>] e.g., #=> ["tftp.service", "tftp.socket"]
+    def keywords
+      keywords = [service.id]
+      keywords << socket.id if socket
+      keywords
+    end
+
+    # Sets the service to be started after calling {#save}
+    #
+    # @see #active=
+    #
+    # @return [Symbol] :start
```

Comment:
Why is the return value important? I would think it is accidental, and not document it, or explicitly documented a [void]

Reply:
Well, in public methods I always like documenting the return value. But yes, I think it is better to use void in this case.